frontend/components/FeedbackBlock/SubscriptionForm.js

```diff
@@ -0,0 +1,154 @@
+import { forwardRef, useReducer, useRef } from "react";
+import PropTypes from "prop-types";
+
+import { MailIcon } from "@heroicons/react/outline";
+
+import ButtonLoadingIcon from "./ButtonLoadingIcon";
+import HolonButton from "../Buttons/HolonButton";
+
+/**
+ * Contains a label and input field. Inputs where `required` is not truthy will have "Optional"
+ * added to the label.
+ */
+// function Input({ label, ...inputProps }) {
+const Input = forwardRef(({ label, ...inputProps }, ref) => (
+  <label className="block pb-3">
+    <span className="block items-baseline pb-1 text-base">
+      <span className="font-semibold">{label}</span>
+      {!inputProps.required && (
+        <span className="ml-3 text-sm text-holon-slated-blue-300">optioneel</span>
+      )}
+    </span>
+    <input
+      {...inputProps}
+      ref={ref}
+      className="w-full rounded-md border-2 border-white bg-transparent p-2 text-base shadow-inner outline-none transition focus:bg-holon-blue-500/50 disabled:opacity-50"
+    />
+  </label>
+));
+
+Input.displayName = "Input";
+
+Input.propTypes = {
+  label: PropTypes.string.isRequired,
+};
+
+/**
+ * Submit button for the form sets its style and text based on the state of the form.
+ */
+function SubmitButton({ status, ...buttonProps }) {
+  let content = "Bevestig";
+
+  if (status === "pending") {
+    content = <ButtonLoadingIcon />;
+  } else if (status === "success") {
+    content = "Met dank!";
+  }
+
+  return (
+    <HolonButton
+      className="flex h-12 items-center justify-center text-lg"
+      disabled={status === "pending" || status === "success"}
+    >
+      {content}
+    </HolonButton>
+  );
+}
+
+SubmitButton.propTypes = {
+  status: PropTypes.oneOf(["initial", "pending", "success", "failed"]).isRequired,
+};
+
+function reducer(state, action) {
+  switch (action.type) {
+    case "fetch":
+      return { status: "pending" };
+    case "resolve":
+      return { status: "success" };
+    case "reject":
+      return { status: "failed" };
+    default:
+      return state;
+  }
+}
+
+export default function SubscriptionForm() {
+  const [state, dispatch] = useReducer(reducer, { status: "initial" });
+
+  const nameRef = useRef();
+  const emailRef = useRef();
+  const companyRef = useRef();
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+
+    dispatch({ type: "fetch" });
+
+    const payload = {
+      name: nameRef.current.value,
+      email: emailRef.current.value,
+      company: companyRef.current.value,
+    };
+
+    fetch(`${process.env.NEXT_PUBLIC_BACKEND_URL}/update-registrations/`, {
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      method: "POST",
+      body: JSON.stringify(payload),
+    })
+      .then(() => dispatch({ type: "resolve" }))
+      .catch(() => dispatch({ type: "reject" }));
```

Comment:
@antw  The following does not work because fetch resolves if the api sends a 400 or 404 or because it is technically an ok response. Please see the following [article](https://towardsdev.com/how-to-handle-404-500-and-more-using-fetch-api-in-javascript-f4e301925a51) for an example on how to catch server errors with fetch.

Also on reject the error that is send by the server is not shown to the user, for example this form can be submitted but does return an error which is not shown.

![image](https://user-images.githubusercontent.com/32770108/177348446-90ad5e4c-9e47-4ae0-9b63-8f876e3b2f72.png)

Reply:
@antw @mattijsstam I will resolve this issue. I will use the package axios, which solve this problem.

Reply:
@antw I see that you already fixed this issue. Thanks. I would have suggest to use the package Axios, but this is good for now I think :)